packages/graphql/src/schema/resolvers/root-connection.ts

```diff
@@ -0,0 +1,124 @@
+ */
+
+import { GraphQLResolveInfo } from "graphql";
+import { InputTypeComposer, SchemaComposer } from "graphql-compose";
+import { PageInfo } from "graphql-relay";
+import { execute } from "../../utils";
+import { translateRead } from "../../translate";
+import { Node } from "../../classes";
+import { Context } from "../../types";
+import getNeo4jResolveTree from "../../utils/get-neo4j-resolve-tree";
+import { isNeoInt } from "../../utils/utils";
+import { createConnectionWithEdgeProperties } from "../pagination";
+
+export default function rootConnectionResolver({ node, composer }: { node: Node; composer: SchemaComposer }) {
+    async function resolve(_root: any, _args: any, _context: unknown, info: GraphQLResolveInfo) {
+        const context = _context as Context;
+        const resolveTree = getNeo4jResolveTree(info);
+
+        const edgeTree = resolveTree.fieldsByTypeName[`${node.name}Connection`].edges;
+        const nodeTree = edgeTree.fieldsByTypeName[`${node.name}Edge`].node;
+
+        context.resolveTree = { ...nodeTree, args: resolveTree.args };
+        context.isRootConnectionField = true;
+
+        const [cypher, params] = translateRead({ context, node });
+
+        const executeResult = await execute({
+            cypher,
+            params,
+            defaultAccessMode: "READ",
+            context,
+        });
+
+        // TODO: Ask neo4j team how to resolve this issue
+        // TCK Tests Run the Resolver But Expect a Return Value
+        // So in order to get the tests to pass we have to return
+        // an empty connection. But it may be more preferable to
+        // throw an error if there is no record, but that might
+        // require modification to the "translateQuery" function
+        let totalCount = 0;
+        let edges: any[] = [];
+        let pageInfo: PageInfo = {
+            hasNextPage: false,
+            hasPreviousPage: false,
+            startCursor: null,
+            endCursor: null,
+        };
+
+        if (executeResult.records[0]) {
+            const record = executeResult.records[0].this;
+
+            totalCount = isNeoInt(record.totalCount) ? record.totalCount.toNumber() : record.totalCount;
+
+            const connection = createConnectionWithEdgeProperties({
+                selectionSet: resolveTree,
+                source: { edges: record.edges },
+                args: resolveTree.args,
+                totalCount,
+            });
+
+            edges = connection.edges as any[];
+            pageInfo = connection.pageInfo as PageInfo;
+        }
+
+        return {
+            totalCount,
+            edges,
+            pageInfo,
+        };
+    }
+
+    const rootEdge = composer.createObjectTC({
+        name: `${node.name}Edge`,
+        fields: {
+            cursor: "String!",
+            node: `${node.name}!`,
+        },
+    });
+
+    const rootConnection = composer.createObjectTC({
+        name: `${node.name}Connection`,
```

Comment:
Pluralize the connection name perhaps? Given that one connection contains multiple edges and thus multiple nodes.

This is also then consistent with relationships, for example `ActorMoviesConnection`.

```suggestion
        name: `${node.plural}Connection`,
```